numpy/lib/arraysetops.py

```diff
@@ -621,28 +621,28 @@ def in1d(ar1, ar2, assume_unique=False, invert=False, *, kind=None):
     # Ensure that iteration through object arrays yields size-1 arrays
     if ar2.dtype == object:
         ar2 = ar2.reshape(-1, 1)
-    # Convert booleans to uint8 so we can use the fast integer algorithm
-    if ar1.dtype == bool:
-        ar1 = ar1 + np.uint8(0)
-    if ar2.dtype == bool:
-        ar2 = ar2 + np.uint8(0)
-
-    # Check if we can use a fast integer algorithm:
-    integer_arrays = (np.issubdtype(ar1.dtype, np.integer) and
-                      np.issubdtype(ar2.dtype, np.integer))
 
     if kind not in {None, 'sort', 'table'}:
         raise ValueError(
             f"Invalid kind: '{kind}'. Please use None, 'sort' or 'table'.")
 
-    if integer_arrays and kind in {None, 'table'}:
+    # Check if we can use a fast integer algorithm:
+    is_integer_arrays = all(ar.dtype.kind in "uib?" for ar in (ar1, ar2))
```

Comment:
Seems the kind is "b" not "?" (that is the char). Having both doesn't make sense.

Reply:
I will assume you mean just delete the `"?"`. Let me know if this isn't correct